pkg/attestation/crafter/materials/oci_image.go

```diff
@@ -281,3 +295,204 @@ func (i *OCIImageCrafter) isLatestTag(ref name.Reference, currentDigest string)
+	// Check if path exists and is a directory
+	info, err := os.Stat(path)
+	if err != nil || !info.IsDir() {
+		return false
+	}
+
+	// Check for oci-layout file
+	layoutFile := filepath.Join(path, ociLayoutRepoName)
+	if _, err := os.Stat(layoutFile); err != nil {
+		return false
+	}
+
+	return true
+}
+
+// craftFromLayout creates a material from an OCI layout directory.
+// If digestSelector is provided, it will look for that specific digest in the layout.
+// Otherwise, it uses the first manifest in the index.
+func (i *OCIImageCrafter) craftFromLayout(_ context.Context, layoutPath, digestSelector string) (*api.Attestation_Material, error) {
+	// Read the OCI layout
+	layoutPath, err := filepath.Abs(layoutPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get absolute path: %w", err)
+	}
+
+	path, err := layout.FromPath(layoutPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read OCI layout: %w", err)
+	}
+
+	// Get the image index
+	index, err := path.ImageIndex()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read image index: %w", err)
+	}
+
+	indexManifest, err := index.IndexManifest()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read index manifest: %w", err)
+	}
+
+	if len(indexManifest.Manifests) == 0 {
+		return nil, fmt.Errorf("no manifests found in OCI layout")
+	}
+
+	// Select the manifest based on digest selector
+	// If a specific digest is requested, find it
+	if digestSelector != "" {
+		found := false
+		var manifest v1.Descriptor
+		for _, m := range indexManifest.Manifests {
+			if m.Digest.String() == digestSelector {
+				manifest = m
+				found = true
+				break
+			}
+		}
+		if !found {
+			return nil, fmt.Errorf("digest %s not found in OCI layout", digestSelector)
+		}
+		i.logger.Debug().Str("digest", digestSelector).Msg("selected image by digest")
+
+		return i.buildMaterialFromManifest(layoutPath, manifest, indexManifest.Manifests)
+	}
+
+	// No digest specified - if multiple images exist, require explicit selection
+	if len(indexManifest.Manifests) > 1 {
+		var digests []string
+		for _, m := range indexManifest.Manifests {
+			digests = append(digests, m.Digest.String())
+		}
+		return nil, fmt.Errorf("OCI layout contains %d images, please specify which one to use with @digest. Available digests: %s",
+			len(indexManifest.Manifests), strings.Join(digests, ", "))
+	}
+
+	// Only one image, safe to use it
+	manifest := indexManifest.Manifests[0]
+	i.logger.Debug().Msg("using only image in layout")
+
+	return i.buildMaterialFromManifest(layoutPath, manifest, indexManifest.Manifests)
+}
+
+// buildMaterialFromManifest constructs the attestation material from a manifest descriptor.
+func (i *OCIImageCrafter) buildMaterialFromManifest(layoutPath string, manifest v1.Descriptor, allManifests []v1.Descriptor) (*api.Attestation_Material, error) {
+
+	digest := manifest.Digest.String()
+
+	// Extract repository name from annotations if available
+	repoName := ociLayoutRepoName + ":"
```

Comment:
as you mentioned before, I'd personally add `unknown` as fallback, that way we always have two parts.